pkg/resource/deploy/source_query.go

```diff
@@ -289,13 +373,66 @@ func (rm *queryResmon) SupportsFeature(ctx context.Context,
 	req *pulumirpc.SupportsFeatureRequest) (*pulumirpc.SupportsFeatureResponse, error) {
 
 	hasSupport := false
-
-	switch req.Id {
-	case "secrets":
-		hasSupport = true
-	}
-
 	return &pulumirpc.SupportsFeatureResponse{
 		HasSupport: hasSupport,
 	}, nil
 }
+
+func (rm *queryResmon) parseProviderRequest(pkg tokens.Package, version string) (providers.ProviderRequest, error) {
```

Comment:
Are these copied from the normal resource monitor? If so, can we refactor them to share the code?

Reply:
We can.